src/App.jsx

```diff
@@ -14,45 +16,95 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+const SORT_FIELD_ALPHA = 'alpha';
+const SORT_FIELD_LENGTH = 'length';
+
+function getPreparedGoods(goods, { sortField, reverseField }) {
+  const preparedGoods = [...goods];
+
+  if (sortField) {
+    preparedGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case SORT_FIELD_ALPHA:
+          return good1.localeCompare(good2);
+
+        case SORT_FIELD_LENGTH:
+          return good1.length - good2.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (reverseField) {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+export const App = () => {
+  const [sortField, setSortField] = useState('');
+  const [reverseField, setReverseField] = useState(false);
+  const visibleGoods = getPreparedGoods(goodsFromServer,
+    { sortField, reverseField });
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          onClick={() => setSortField(SORT_FIELD_ALPHA)}
+          type="button"
+          className={cn('button', 'is-info',
+            {
+              'is-light': sortField !== SORT_FIELD_ALPHA,
+            })}
+        >
+          Sort alphabetically
+        </button>
+
+        <button
+          onClick={() => setSortField(SORT_FIELD_LENGTH)}
+          type="button"
+          className={cn('button', 'is-success',
+            {
+              'is-light': sortField !== SORT_FIELD_LENGTH,
+            })}
+        >
+          Sort by length
+        </button>
+
+        <button
+          onClick={() => setReverseField(!reverseField)}
```

Comment:
and move this click handler to an own function
```suggestion
          onClick={() => setReverseField(prev => !prev)}
```

Reply:
isn't fixed.
You should move onClick handler to an own function (variable) and use here
So you create a variable like `handleReverseClick` at 52th row and pass this function to onClick onClick={handleReverseClick}